src/debugAdapter/goDebug.ts

```diff
@@ -313,29 +313,21 @@ class Delve {
 			});
 
 			function connectClient(port: number, host: string) {
-				// Add a slight delay to avoid issues on Linux with
-				// Delve failing calls made shortly after connection.
-				setTimeout(() => {
-					let client = Client.$create(port, host);
-					client.connectSocket((err, conn) => {
-						if (err) return reject(err);
-						return resolve(conn);
-					});
-				}, 200);
+				let client = Client.$create(port, host);
+				client.connectSocket((err, conn) => {
+					if (err) return reject(err);
+					return resolve(conn);
+				});
 			}
 
 			this.debugProcess.stderr.on('data', chunk => {
 				let str = chunk.toString();
 				if (this.onstderr) { this.onstderr(str); }
-				if (!serverRunning) {
-					serverRunning = true;
-					connectClient(port, host);
-				}
 			});
 			this.debugProcess.stdout.on('data', chunk => {
 				let str = chunk.toString();
 				if (this.onstdout) { this.onstdout(str); }
-				if (!serverRunning) {
+				if (!serverRunning && str.startsWith('API server listening at: ')) {
```

Comment:
@derekparker Is there any other recommended way for editors to figure out when to connect the client to `dlv` other than waiting for the message `API server listening at:` on the stdout?

Reply:
The editor should know the host / port it is trying to connect to, would it be possible to simply retry (potentially w/ backoff, or an upper-bound retry time) the connection until the server is running and accepting connections?

Reply:
Yes, that's possible and has come up as one of the solutions when I was discussing with @roblourens 
Thanks @derekparker